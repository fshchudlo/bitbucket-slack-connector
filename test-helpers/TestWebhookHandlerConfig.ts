import { WebhookHandlerConfig } from "../bitbucket-webhook-handler/webhookHandlerConfig";

export const TestWebhookHandlerConfig: WebhookHandlerConfig = {
    getOpenedPRBroadcastChannelId(): string | null {
        return null;
    },
    usePrivateChannels: true
};