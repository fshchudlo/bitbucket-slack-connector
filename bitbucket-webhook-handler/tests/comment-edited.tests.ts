import SlackAdapterSnapshottingMock from "./mocks/SlackAdapterSnapshottingMock";
import TestPayloadBuilder from "./mocks/TestPayloadBuilder";
import handleBitbucketWebhook from "../handleBitbucketWebhook";
import { MockBitbucketAPIAdapter } from "./mocks/MockBitbucketAPIAdapter";
import { TestWebhookHandlerConfig } from "./mocks/TestWebhookHandlerConfig";

describe("handleBitbucketWebhook", () => {

    it("Should send message on PR comment edit", async () => {
        const testSlackGateway = await new SlackAdapterSnapshottingMock().setupBasicChannel();
        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestCommentAdded(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        const payload = TestPayloadBuilder.pullRequestCommentEdited();
        await handleBitbucketWebhook(payload, testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        expect(testSlackGateway.snapshot).toMatchSnapshot();
    });

    it("Should send message on comment conversion to the task", async () => {
        const testSlackGateway = await new SlackAdapterSnapshottingMock().setupBasicChannel();
        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestCommentAdded(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        const payload = TestPayloadBuilder.pullRequestCommentConvertedToTheTask();
        await handleBitbucketWebhook(payload, testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        expect(testSlackGateway.snapshot).toMatchSnapshot();
    });

    it("Should send message on task conversion to the comment", async () => {
        const testSlackGateway = await new SlackAdapterSnapshottingMock().setupBasicChannel();
        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestTaskAdded(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        const payload = TestPayloadBuilder.pullRequestTaskConvertedToTheComment();
        await handleBitbucketWebhook(payload, testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        expect(testSlackGateway.snapshot).toMatchSnapshot();
    });
    it("Should send message on comment resolving and reopening", async () => {
        const testSlackGateway = await new SlackAdapterSnapshottingMock().setupBasicChannel();
        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestCommentAdded(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestCommentResolved(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);
        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestCommentReopened(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        expect(testSlackGateway.snapshot).toMatchSnapshot();
    });

    it("Should send message on task resolving and reopening", async () => {
        const testSlackGateway = await new SlackAdapterSnapshottingMock().setupBasicChannel();
        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestTaskAdded(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestTaskResolved(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);
        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestTaskReopened(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        expect(testSlackGateway.snapshot).toMatchSnapshot();
    });

    it("Should return generic message if initial comment was not found", async () => {
        const testSlackGateway = await new SlackAdapterSnapshottingMock().setupBasicChannel();


        await handleBitbucketWebhook(TestPayloadBuilder.pullRequestTaskReopened(), testSlackGateway, new MockBitbucketAPIAdapter(), TestWebhookHandlerConfig);


        expect(testSlackGateway.snapshot).toMatchSnapshot();
    });
});
