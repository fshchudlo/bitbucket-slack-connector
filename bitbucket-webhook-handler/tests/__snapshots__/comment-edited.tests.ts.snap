// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleBitbucketWebhook Should return generic message if initial comment was not found 1`] = `
{
  "addedBookmarks": [
    {
      "channelId": "12345",
      "emoji": ":git:",
      "link": "https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview",
      "title": "Review Pull Request",
    },
  ],
  "addedReactions": [],
  "archivedChannels": [],
  "createdChannels": [
    {
      "id": "12345",
      "isArchived": false,
      "name": "pr-test-proj-test-repository-1",
    },
  ],
  "invitesToChannels": [
    {
      "channelId": "12345",
      "force": true,
      "users": [
        "test.author@test.com",
        "test.reviewer1@test.com",
        "test.reviewer2@test.com",
      ],
    },
  ],
  "kicksFromChannels": [],
  "lookedUpUsers": [
    [
      {
        "displayName": "Test Author",
        "emailAddress": "test.author@test.com",
        "name": "TestAuthor",
      },
      {
        "displayName": "Test Reviewer 1",
        "emailAddress": "test.reviewer1@test.com",
        "name": "TestReviewer1",
      },
      {
        "displayName": "Test Reviewer 2",
        "emailAddress": "test.reviewer2@test.com",
        "name": "TestReviewer2",
      },
    ],
  ],
  "searchedChannels": [
    "pr-test-proj-test-repository-1",
  ],
  "searchedCommentSnapshots": [
    {
      "bitbucketCommentId": 1,
      "channelId": "12345",
    },
  ],
  "searchedPrOpenedBroadcastMessages": [],
  "sentMessages": [
    {
      "blocks": [
        {
          "text": {
            "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "type": "divider",
        },
        {
          "elements": [
            {
              "text": "Test pull request description",
              "type": "mrkdwn",
            },
          ],
          "type": "context",
        },
        {
          "type": "divider",
        },
        {
          "text": {
            "text": "Please <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|review the PR>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":writing_hand: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|edited task>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "BLOCKER",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": undefined,
      "text": ":writing_hand: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|edited task>:",
      "threadId": undefined,
    },
  ],
}
`;

exports[`handleBitbucketWebhook Should send message on PR comment edit 1`] = `
{
  "addedBookmarks": [
    {
      "channelId": "12345",
      "emoji": ":git:",
      "link": "https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview",
      "title": "Review Pull Request",
    },
  ],
  "addedReactions": [],
  "archivedChannels": [],
  "createdChannels": [
    {
      "id": "12345",
      "isArchived": false,
      "name": "pr-test-proj-test-repository-1",
    },
  ],
  "invitesToChannels": [
    {
      "channelId": "12345",
      "force": true,
      "users": [
        "test.author@test.com",
        "test.reviewer1@test.com",
        "test.reviewer2@test.com",
      ],
    },
  ],
  "kicksFromChannels": [],
  "lookedUpUsers": [
    [
      {
        "displayName": "Test Author",
        "emailAddress": "test.author@test.com",
        "name": "TestAuthor",
      },
      {
        "displayName": "Test Reviewer 1",
        "emailAddress": "test.reviewer1@test.com",
        "name": "TestReviewer1",
      },
      {
        "displayName": "Test Reviewer 2",
        "emailAddress": "test.reviewer2@test.com",
        "name": "TestReviewer2",
      },
    ],
  ],
  "searchedChannels": [
    "pr-test-proj-test-repository-1",
    "pr-test-proj-test-repository-1",
  ],
  "searchedCommentSnapshots": [
    {
      "bitbucketCommentId": 1,
      "channelId": "12345",
    },
  ],
  "searchedPrOpenedBroadcastMessages": [],
  "sentMessages": [
    {
      "blocks": [
        {
          "text": {
            "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "type": "divider",
        },
        {
          "elements": [
            {
              "text": "Test pull request description",
              "type": "mrkdwn",
            },
          ],
          "type": "context",
        },
        {
          "type": "divider",
        },
        {
          "text": {
            "text": "Please <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|review the PR>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added comment>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "NORMAL",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": undefined,
      "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added comment>:",
      "threadId": undefined,
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":writing_hand: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|edited comment>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Updated comment text",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "NORMAL",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": true,
      "text": ":writing_hand: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|edited comment>:",
      "threadId": "ABCDE",
    },
  ],
}
`;

exports[`handleBitbucketWebhook Should send message on comment conversion to the task 1`] = `
{
  "addedBookmarks": [
    {
      "channelId": "12345",
      "emoji": ":git:",
      "link": "https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview",
      "title": "Review Pull Request",
    },
  ],
  "addedReactions": [],
  "archivedChannels": [],
  "createdChannels": [
    {
      "id": "12345",
      "isArchived": false,
      "name": "pr-test-proj-test-repository-1",
    },
  ],
  "invitesToChannels": [
    {
      "channelId": "12345",
      "force": true,
      "users": [
        "test.author@test.com",
        "test.reviewer1@test.com",
        "test.reviewer2@test.com",
      ],
    },
  ],
  "kicksFromChannels": [],
  "lookedUpUsers": [
    [
      {
        "displayName": "Test Author",
        "emailAddress": "test.author@test.com",
        "name": "TestAuthor",
      },
      {
        "displayName": "Test Reviewer 1",
        "emailAddress": "test.reviewer1@test.com",
        "name": "TestReviewer1",
      },
      {
        "displayName": "Test Reviewer 2",
        "emailAddress": "test.reviewer2@test.com",
        "name": "TestReviewer2",
      },
    ],
  ],
  "searchedChannels": [
    "pr-test-proj-test-repository-1",
    "pr-test-proj-test-repository-1",
  ],
  "searchedCommentSnapshots": [
    {
      "bitbucketCommentId": 1,
      "channelId": "12345",
    },
  ],
  "searchedPrOpenedBroadcastMessages": [],
  "sentMessages": [
    {
      "blocks": [
        {
          "text": {
            "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "type": "divider",
        },
        {
          "elements": [
            {
              "text": "Test pull request description",
              "type": "mrkdwn",
            },
          ],
          "type": "context",
        },
        {
          "type": "divider",
        },
        {
          "text": {
            "text": "Please <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|review the PR>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added comment>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "NORMAL",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": undefined,
      "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added comment>:",
      "threadId": undefined,
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":pushpin: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|converted comment to the task>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "BLOCKER",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": true,
      "text": ":pushpin: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|converted comment to the task>:",
      "threadId": "ABCDE",
    },
  ],
}
`;

exports[`handleBitbucketWebhook Should send message on comment resolving and reopening 1`] = `
{
  "addedBookmarks": [
    {
      "channelId": "12345",
      "emoji": ":git:",
      "link": "https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview",
      "title": "Review Pull Request",
    },
  ],
  "addedReactions": [],
  "archivedChannels": [],
  "createdChannels": [
    {
      "id": "12345",
      "isArchived": false,
      "name": "pr-test-proj-test-repository-1",
    },
  ],
  "invitesToChannels": [
    {
      "channelId": "12345",
      "force": true,
      "users": [
        "test.author@test.com",
        "test.reviewer1@test.com",
        "test.reviewer2@test.com",
      ],
    },
  ],
  "kicksFromChannels": [],
  "lookedUpUsers": [
    [
      {
        "displayName": "Test Author",
        "emailAddress": "test.author@test.com",
        "name": "TestAuthor",
      },
      {
        "displayName": "Test Reviewer 1",
        "emailAddress": "test.reviewer1@test.com",
        "name": "TestReviewer1",
      },
      {
        "displayName": "Test Reviewer 2",
        "emailAddress": "test.reviewer2@test.com",
        "name": "TestReviewer2",
      },
    ],
  ],
  "searchedChannels": [
    "pr-test-proj-test-repository-1",
    "pr-test-proj-test-repository-1",
    "pr-test-proj-test-repository-1",
  ],
  "searchedCommentSnapshots": [
    {
      "bitbucketCommentId": 1,
      "channelId": "12345",
    },
    {
      "bitbucketCommentId": 1,
      "channelId": "12345",
    },
  ],
  "searchedPrOpenedBroadcastMessages": [],
  "sentMessages": [
    {
      "blocks": [
        {
          "text": {
            "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "type": "divider",
        },
        {
          "elements": [
            {
              "text": "Test pull request description",
              "type": "mrkdwn",
            },
          ],
          "type": "context",
        },
        {
          "type": "divider",
        },
        {
          "text": {
            "text": "Please <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|review the PR>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added comment>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "NORMAL",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": undefined,
      "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added comment>:",
      "threadId": undefined,
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":white_check_mark: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|resolved comment>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "NORMAL",
          "taskResolvedDate": undefined,
          "threadResolvedDate": 123456789,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": true,
      "text": ":white_check_mark: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|resolved comment>:",
      "threadId": "ABCDE",
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":repeat: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|reopened comment>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "NORMAL",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": true,
      "text": ":repeat: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|reopened comment>:",
      "threadId": "ABCDE",
    },
  ],
}
`;

exports[`handleBitbucketWebhook Should send message on task conversion to the comment 1`] = `
{
  "addedBookmarks": [
    {
      "channelId": "12345",
      "emoji": ":git:",
      "link": "https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview",
      "title": "Review Pull Request",
    },
  ],
  "addedReactions": [],
  "archivedChannels": [],
  "createdChannels": [
    {
      "id": "12345",
      "isArchived": false,
      "name": "pr-test-proj-test-repository-1",
    },
  ],
  "invitesToChannels": [
    {
      "channelId": "12345",
      "force": true,
      "users": [
        "test.author@test.com",
        "test.reviewer1@test.com",
        "test.reviewer2@test.com",
      ],
    },
  ],
  "kicksFromChannels": [],
  "lookedUpUsers": [
    [
      {
        "displayName": "Test Author",
        "emailAddress": "test.author@test.com",
        "name": "TestAuthor",
      },
      {
        "displayName": "Test Reviewer 1",
        "emailAddress": "test.reviewer1@test.com",
        "name": "TestReviewer1",
      },
      {
        "displayName": "Test Reviewer 2",
        "emailAddress": "test.reviewer2@test.com",
        "name": "TestReviewer2",
      },
    ],
  ],
  "searchedChannels": [
    "pr-test-proj-test-repository-1",
    "pr-test-proj-test-repository-1",
  ],
  "searchedCommentSnapshots": [
    {
      "bitbucketCommentId": 1,
      "channelId": "12345",
    },
  ],
  "searchedPrOpenedBroadcastMessages": [],
  "sentMessages": [
    {
      "blocks": [
        {
          "text": {
            "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "type": "divider",
        },
        {
          "elements": [
            {
              "text": "Test pull request description",
              "type": "mrkdwn",
            },
          ],
          "type": "context",
        },
        {
          "type": "divider",
        },
        {
          "text": {
            "text": "Please <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|review the PR>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added task>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "BLOCKER",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": undefined,
      "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added task>:",
      "threadId": undefined,
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":writing_hand: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|converted task to the comment>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "NORMAL",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": true,
      "text": ":writing_hand: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|converted task to the comment>:",
      "threadId": "ABCDE",
    },
  ],
}
`;

exports[`handleBitbucketWebhook Should send message on task resolving and reopening 1`] = `
{
  "addedBookmarks": [
    {
      "channelId": "12345",
      "emoji": ":git:",
      "link": "https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview",
      "title": "Review Pull Request",
    },
  ],
  "addedReactions": [],
  "archivedChannels": [],
  "createdChannels": [
    {
      "id": "12345",
      "isArchived": false,
      "name": "pr-test-proj-test-repository-1",
    },
  ],
  "invitesToChannels": [
    {
      "channelId": "12345",
      "force": true,
      "users": [
        "test.author@test.com",
        "test.reviewer1@test.com",
        "test.reviewer2@test.com",
      ],
    },
  ],
  "kicksFromChannels": [],
  "lookedUpUsers": [
    [
      {
        "displayName": "Test Author",
        "emailAddress": "test.author@test.com",
        "name": "TestAuthor",
      },
      {
        "displayName": "Test Reviewer 1",
        "emailAddress": "test.reviewer1@test.com",
        "name": "TestReviewer1",
      },
      {
        "displayName": "Test Reviewer 2",
        "emailAddress": "test.reviewer2@test.com",
        "name": "TestReviewer2",
      },
    ],
  ],
  "searchedChannels": [
    "pr-test-proj-test-repository-1",
    "pr-test-proj-test-repository-1",
    "pr-test-proj-test-repository-1",
  ],
  "searchedCommentSnapshots": [
    {
      "bitbucketCommentId": 1,
      "channelId": "12345",
    },
    {
      "bitbucketCommentId": 1,
      "channelId": "12345",
    },
  ],
  "searchedPrOpenedBroadcastMessages": [],
  "sentMessages": [
    {
      "blocks": [
        {
          "text": {
            "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "type": "divider",
        },
        {
          "elements": [
            {
              "text": "Test pull request description",
              "type": "mrkdwn",
            },
          ],
          "type": "context",
        },
        {
          "type": "divider",
        },
        {
          "text": {
            "text": "Please <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|review the PR>",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "text": "Test Author opened <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview|pull request>",
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added task>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "BLOCKER",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": undefined,
      "text": ":loudspeaker: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|added task>:",
      "threadId": undefined,
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":white_check_mark: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|resolved task>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "BLOCKER",
          "taskResolvedDate": 123456789,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": true,
      "text": ":white_check_mark: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|resolved task>:",
      "threadId": "ABCDE",
    },
    {
      "blocks": [
        {
          "text": {
            "text": ":repeat: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|reopened task>:",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        {
          "text": {
            "text": "> Test comment",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "channelId": "12345",
      "iconEmoji": ":bitbucket:",
      "metadata": {
        "eventPayload": {
          "commentId": "1",
          "commentParentId": undefined,
          "severity": "BLOCKER",
          "taskResolvedDate": undefined,
          "threadResolvedDate": undefined,
        },
        "eventType": "bitbucket_comment_snapshot_saved",
      },
      "replyBroadcast": true,
      "text": ":repeat: Test Reviewer 1 <https://git.test.com/projects/TEST-PROJ/repos/test-repository/pull-requests/1/overview?commentId=1|reopened task>:",
      "threadId": "ABCDE",
    },
  ],
}
`;
